src/wh_server.c

```diff
@@ -53,26 +53,53 @@ static int _wh_Server_HandleCustomRequest(whServerContext* server,
 int wh_Server_Init(whServerContext* server, whServerConfig* config)
 {
     int rc = 0;
+
     if ((server == NULL) || (config == NULL)) {
         return WH_ERROR_BADARGS;
     }
 
     memset(server, 0, sizeof(*server));
-    if (
-        ((rc = wolfCrypt_Init()) == 0) &&
-        ((rc = wc_InitRng_ex(server->crypto->rng, NULL, INVALID_DEVID)) == 0)) {
-        rc = wh_Nvm_Init(server->nvm, config->nvm_config);
-        if (rc == 0) {
-            server->nvm->cb = config->nvm_config->cb;
-            server->nvm->context = config->nvm_config->context;
 
-            rc = wh_CommServer_Init(server->comm, config->comm_config);
-            if (rc == 0) {
-                /* All good */
-            }
+    rc = wolfCrypt_Init();
+    if (rc != 0) {
+        (void)wh_Server_Cleanup(server);
+        return WH_ERROR_ABORTED;
+    }
+
+#if defined(WOLF_CRYPTO_CB)
+    server->crypto->devId = config->devId;
+    if (config->cryptocb != NULL) {
+        /* register the crypto callback with wolSSL */
+        rc = wc_CryptoCb_RegisterDevice(server->crypto->devId,
+                                        config->cryptocb,
+                                        NULL);
+        if (rc != 0) {
+            (void)wh_Server_Cleanup(server);
+            return WH_ERROR_ABORTED;
         }
-    } else {
-        wh_Server_Cleanup(server);
+    }
+#else
+    server->crypto->devId = INVALID_DEVID;
+#endif
+
+    rc = wc_InitRng_ex(server->crypto->rng, NULL, server->crypto->devId);
+    if (rc != 0) {
+        (void)wh_Server_Cleanup(server);
+        return WH_ERROR_ABORTED;
+    }
+
+    rc = wh_Nvm_Init(server->nvm, config->nvm_config);
+    if (rc != 0) {
+        (void)wh_Server_Cleanup(server);
+        return WH_ERROR_ABORTED;
+    }
+    server->nvm->cb      = config->nvm_config->cb;
+    server->nvm->context = config->nvm_config->context;
```

Comment:
This should have already been moved into Nvm_Init.  I should have removed these lines.  my bad.

Reply:
done